### ISSUE TITLE
Bug: Restarting or poweroff sometimes fails

### ISSUE THREAD

**Issue by likeadragonmaid** [ubports/ubuntu-touch#2046]
- Device: Mi A2
- Channel: devel
- Build: 2022-09-05

### Steps to reproduce

1. Use power button to show the power menu
2. Press power off or restart

### Expected behavior
Phone should power off if power off option is pressed. Phone should restart if restart option is pressed.

### Actual behavior
When the bug appears, Lomiri shell simply restarts and brings you back to lockscreen and phone stays on. However once again trying to power off or restart the device results in successful power off or restart.

### Logfiles and additional information
[unity8.log.txt](https://github.com/ubports/ubuntu-touch/files/9496202/unity8.log.txt)

**Comment by Flohack74** [comment]
Yeah its basically an out of memory situation so far. Did you have many Apps open when that happens?

**Comment by likeadragonmaid** [comment]
I might have opened 2-3 apps but closed them before trying to turn the phone off.
